### PR TITLE
Support user-defined XML root node

### LINKED ISSUE #150: Do not hardcode iati-activities for top level for all xml.
https://github.com/OpenDataServices/flatten-tool/blob/master/flattentool/xml_output.py#L28

Would it make sense to be root_list_path or maybe some other special option for xml conversion.

### PR DESCRIPTION
**Pull request by andylolz**
UPDATE: I’ve changed this in e9688d7, to remove the `--xml-root-path` CLI option and `xml_root_path` unflatten arg. Instead, the user can specify a custom root node via an XMLRootPath directive in the metatab. This is more flexible anyway I suppose, since you could probably still use something like:

```bash
./flatten-tool unflatten --default-configuration 'XMLRootPath iati-organisations' […]
```

Refs #176.

---

Fixes #150 by adding a `--xml-root-path` option to the CLI, and `xml_root_path` arg to unflatten.

The default is set to ‘iati-activities’ for backwards compatibility.

A test is added (in a separate commit) to demonstrate that this could be used for an IATI organisation file.

I don’t know how roundtripping this should work. One option would be to remove the CLI option and arg, and instead use a metatab directive for this i.e. parse the following from the top of the meta tab:

```csv
#,XMLRootPath iati-organisations,skiprows 9000
```

I actually prefer that – I think it would be preferable (and would make life easier for CoVE) to do more stuff that way. It’s not like the user sometimes wants to output this data with a different root node. The specific root node is coupled to the rest of the data; it seems to make sense that it should live with it in the flattened file(s). Wdyt @Bjwebb?

If this is merged, I can continue adding IATI org file support to CoVE.

### CODE REVIEW

File: flattentool/__init__.py [hunk 1]
```diff
@@ -208,14 +208,15 @@ def unflatten(input_name, base_json=None, input_format=None, output_name=None,
         base[root_list_path] = list(result)
 
     if xml:
+        xml_root_path = base_configuration.get('XMLRootPath', 'iati-activities')
```
Comment (Bjwebb) [inline flattentool/__init__.py]: I'm wondering whether we should remove this default entirely.

File: flattentool/__init__.py [hunk 2]
```diff
@@ -208,14 +208,15 @@ def unflatten(input_name, base_json=None, input_format=None, output_name=None,
...
         if output_name is None:
             if sys.version > '3':
-                sys.stdout.buffer.write(toxml(base))
+                sys.stdout.buffer.write(toxml(base, xml_root_path))
```
Comment (Bjwebb) [inline flattentool/__init__.py]: Could we call this `xml_root_tag` instead, as it's not really a path?

File: flattentool/lib.py [hunk 3]
```diff
@@ -17,4 +17,6 @@ def parse_sheet_configuration(configuration_list):
             configuration['ignore'] = True
         if (len(parts) == 1 and parts[0].lower() in ("hashcomments", "hashcomment")):
             configuration['hashcomments'] = True
+        if (len(parts) == 2 and parts[0].lower() == "xmlrootpath"):
+            configuration['XMLRootPath'] = parts[1]
```
Comment (Bjwebb) [inline flattentool/lib.py]: Could you also add one of these for RootListPath also?

File: flattentool/__init__.py (hunk 2 above)
Comment (andylolz) [reply flattentool/__init__.py]: sure!

File: flattentool/lib.py (hunk 3 above)
Comment (andylolz) [reply flattentool/lib.py]: Sure!

File: flattentool/__init__.py (hunk 1 above)
Comment (andylolz) [reply flattentool/__init__.py]: Ohhhhh… In fact, it’s actually a bit tricky and relies on #176, I think. (That’s possibly why I reffed #176 in the PR, I don’t remember.)

Can we revisit after #176 is sorted?

**Comment by andylolz** [comment]
Okay – requested changes are (hopefully) all addressed @Bjwebb.

File: flattentool/__init__.py (hunk 2 above)
Comment (andylolz) [reply flattentool/__init__.py]: Fixed in 23dcdb4 and cada684.

File: flattentool/lib.py (hunk 3 above)
Comment (andylolz) [reply flattentool/lib.py]: Fixed in 795644f and 769b272.

File: flattentool/__init__.py (hunk 1 above)
Comment (Bjwebb) [reply flattentool/__init__.py]: > Can we revisit after #176 is sorted?

Yes, that makes sense.

**Comment by Bjwebb** [comment]
Looks good, thanks.

**Pull request merged**